intranet/templates/dashboard/dashboard.html

```diff
@@ -206,16 +207,28 @@ <h3>
             </div>
         {% endfor %}
         {% if not request.user.is_restricted %}
-        {% if start_num == 0 and view_announcements_url != 'announcements_archive' %}
+        {% if page_num == 1 and view_announcements_url != 'announcements_archive' %}
             <a href="{% url 'announcements_archive' %}" class="button" style="float:left"><i class="fas fa-archive" style="width: 13px"></i> View Archive</a>
         {% endif %}
 
-        {% if start_num > 0 %}
-            <a href="{% url view_announcements_url %}?start={{ prev_page }}{{ paginate_link_suffix }}" class="button" style="float:left">&larr; Newer Posts</a>
+        {% if prev_page > 0 %}
+            <a href="{% url view_announcements_url %}?page={{ prev_page }}{{ paginate_link_suffix }}" class="button" style="float:left">&larr; Newer Posts</a>
         {% endif %}
         {% if more_items %}
-            <a href="{% url view_announcements_url %}?start={{ end_num }}{{ paginate_link_suffix }}" class="button" style="float:right">Older Posts &rarr;</a>
+            <a href="{% url view_announcements_url %}?page={{ next_page }}{{ paginate_link_suffix }}" class="button" style="float:right">Older Posts &rarr;</a>
         {% endif %}
+        <div style="text-align: center">
+            {% for page in page_obj|page_list:items %}
+            <a {% if page %} class="button" {% else %} class="ellipses" {% endif %}
+              href="{% url view_announcements_url %}{% if page %}?{% query_transform request page=page %}{% else %}#{% endif %}">{{ page|default:". . ." }}</a>
+            {% endfor %}
+            <form action="{% url view_announcements_url %}" method="get">
+                <!--Only allow numbers as the input-->
+                <input name="page" type="text"
+                onkeypress="return /[0-9]/i.test(event.key)" id="announce-page-num" style="width: 75px"
+                placeholder={{ items.number }}> of {{ page_obj.num_pages }}
+            </form>
+        </div>
```

Comment:
This might just be personal preference, but I think the page numbers and input box look a bit out-of-place when there's only one page. Have you considered hiding this div in such a scenario?